### PR TITLE
Add cluster_role/cluster_role_binding support

### PR DESCRIPTION
**Pull request by korjek**

### CODE REVIEW

**Comment by korjek** [comment]
@jtopjian could you please have a look?
Thanx

File: docs/data_sources/sensu_cluster_role.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+```
+
+## Argument Reference
+
+* `name` - *Required* - The name of the Sensu cluster role.
+
+## Attribute Reference
+
+* `rule` - *Required* - See the [Sensu rbac reference](https://docs.sensu.io/sensu-go/latest/reference/rbac/#rule-attributes).
```
Comment (jtopjian) [inline docs/data_sources/sensu_cluster_role.md:21]: Attributes (or outputs) aren't required - we can remove the word `*Required*` here.

File: docs/data_sources/sensu_cluster_role.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+### rule
+
+The `rule` block supports:
+
+* `verbs` - *Required* - See the [Sensu rbac reference](https://docs.sensu.io/sensu-go/latest/reference/rbac/#rule-attributes).
```
Comment (jtopjian) [inline docs/data_sources/sensu_cluster_role.md:27]: same here, since this is part of `rule`.

File: docs/data_sources/sensu_cluster_role.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+* `resources` - *Required* - See the [Sensu rbac reference](https://docs.sensu.io/sensu-go/latest/reference/rbac/#rule-attributes).
```
Comment (jtopjian) [inline docs/data_sources/sensu_cluster_role.md:29]: same here, since this is part of rule.

**Review by jtopjian — CHANGES_REQUESTED** [review]
@korjek Thanks so much for your work on this.

I left a few comments about some minor documentation changes. The other issue is that there appears to be a binary bundled into this PR (`sensu/terraform-provider-sensu` - see at the very bottom of the page on the "Files changed" tab).

I have a question about the design of the role binding resource: you have made explicit arguments for `users` and `groups`, which is easy to understand for the person using this resource. However, I wonder if there is benefit of matching closely to the API by removing users and groups and use "subject"?

```go
"subject": &schema.Schema{
  Type: schema.TypeList,
  Required: true,
  ForceNew: true,
  MinItems: 1,
  Elem: &schema.Resource{
    Schema: map[string]*schema.Schema{
      "name": &schema.Schema{
        Type: schema.TypeString,
        Required: true,
      },
      "type": &schema.Schema{
        Type: schema.TypeString,
        Required: true,
      }.
    }.
  }.
}
```

If we designed it this way, role bindings would be able to automatically support resources beyond users and groups (if that happens in the future).

I am not opposed to the way you have designed this - just raising this for discussion.

Thoughts?

File: docs/data_sources/sensu_cluster_role.md (hunk 3 above)
Comment (korjek) [reply docs/data_sources/sensu_cluster_role.md:29]: hi @jtopjian 
Thank you for your review.

Unfortunately, I missed the point that I opened this PR from my personal account. I would like to close this PR and open it from my company git account.
With this new PR I'll address all your comments except "subject", so we can continue the discussion on the thread.

Sorry for the inconvenience.

Comment (jtopjian) [reply docs/data_sources/sensu_cluster_role.md:29]: @fgouteroux Sure, sounds good. It's no inconvenience at all 🙂

**Pull request closed without merging**